src/boot/efi/meson.build

```diff
@@ -3,7 +3,7 @@
 conf.set10('ENABLE_EFI', get_option('efi'))
 conf.set10('HAVE_GNU_EFI', false)
 
-if not get_option('efi') or get_option('gnu-efi') == 'false'
+if skip_deps or not get_option('efi') or get_option('gnu-efi') == 'false'
```

Comment:
I think this will break the fuzz target I'm trying to add in https://github.com/systemd/systemd/pull/21915 (because on OSS-Fuzz skip-deps is set to true always). Considering that it's the only places where the fuzz targets are run on a regular basis it would mean that that fuzz target could be effectively thrown away

Reply:
The fuzzer can easily be added in front of this check.

Reply:
as far as I understand it should be wrapped in
```
if efi_arch[1] in ['ia32', 'x86_64', 'arm', 'aarch64']
        systemd_boot_sources += 'bcd.c'
        tests += [
                [['src/boot/efi/test-bcd.c'],
                 [],
                 [libzstd],
                 [],
                 'HAVE_ZSTD'],
        ]
endif
```
and at that point it would be complicated

Reply:
Requiring gnu-efi to be around for fuzzing when the fuzzer doesn't even need it is complication.

Reply:
I agree and I wrote a comment in that PR that `gnu-efi` shouldn't have to be installed there to build and run the fuzzer but I think there should be a better way to decouple the bcd parser from "gnu-efi". I mean, I agree it makes sense to skip all that when skip_deps is true and I generally have no problem with adding the fuzz target at the beginning but it seems to be compatible with OSS-Fuzz only and the unit tests would be kind of broken because the fuzz target would be always built and `test-bcd` is built depending on the architecture.

Reply:
Looking at https://github.com/systemd/systemd/pull/21922/files, I think this particular issue could be solved by placing
```
efi_arch = {
        # host_cc_arch: [efi_arch (see Table 3-2 in UEFI spec), gnu_efi_inc_arch]
        'x86':     ['ia32', 'ia32'],
        'x86_64':  ['x64', 'x86_64'],
        'arm':     ['arm', 'arm'],
        'aarch64': ['aa64', 'aarch64'],
        'riscv64': ['riscv64', 'riscv64'],
}.get(host_machine.cpu_family(), [])
```
right before the "if" expression where skip_deps is checked. This way it should be possible to include the fuzz target depending on the architecture by analogy with `test-bcd`.

Reply:
One downside would be that people compiling systemd with `-Dgnu-efi=false -Dfuzz-tests=true` would still get the bcd fuzzer they probably don't care about but I doubt anyone uses this combination in practice

Reply:
With the "fuzz-bcd" branch rebased on top of this PR I changed `meson.build` and `tools/oss-fuzz` to build the fuzzer depending on the architecture and also dropped `apt-get install gnu-efi`:
```diff
diff --git a/src/boot/efi/meson.build b/src/boot/efi/meson.build
index 9e7fbced67..b5f3596fc6 100644
--- a/src/boot/efi/meson.build
+++ b/src/boot/efi/meson.build
@@ -3,13 +3,6 @@
 conf.set10('ENABLE_EFI', get_option('efi'))
 conf.set10('HAVE_GNU_EFI', false)

-if skip_deps or not get_option('efi') or get_option('gnu-efi') == 'false'
-        if get_option('gnu-efi') == 'true'
-                error('gnu-efi support requested, but general efi support is disabled')
-        endif
-        subdir_done()
-endif
-
 efi_arch = {
         # host_cc_arch: [efi_arch (see Table 3-2 in UEFI spec), gnu_efi_inc_arch]
         'x86':     ['ia32', 'ia32'],
@@ -19,6 +12,26 @@ efi_arch = {
         'riscv64': ['riscv64', 'riscv64'],
 }.get(host_machine.cpu_family(), [])

+# The fuzz target is always built regardless of whether gnu-efi is enabled or not
+# because it doesn't depend on gnu-efi. It makes it possible to
+# build and run it when the gnu-efi package isn't installed
+# (like on OSS-Fuzz for example).
+#
+# The fuzzer only makes sense on arches supported by Windows
+# (by analogy with test-bcd).
+if efi_arch[1] in ['ia32', 'x86_64', 'arm', 'aarch64']
+        fuzzers += [
+                [['src/boot/efi/fuzz-bcd.c']],
+        ]
+endif
+
+if skip_deps or not get_option('efi') or get_option('gnu-efi') == 'false'
+        if get_option('gnu-efi') == 'true'
+                error('gnu-efi support requested, but general efi support is disabled')
+        endif
+        subdir_done()
+endif
+
 efi_incdir = get_option('efi-includedir')
 if efi_arch.length() > 0 and not cc.has_header('@0@/@1@/efibind.h'.format(efi_incdir, efi_arch[1]))
         efi_arch = []
diff --git a/tools/oss-fuzz.sh b/tools/oss-fuzz.sh
index 98271dd93b..df9bad9aa1 100755
--- a/tools/oss-fuzz.sh
+++ b/tools/oss-fuzz.sh
@@ -35,13 +35,7 @@ else
     apt-get update
     apt-get install -y gperf m4 gettext python3-pip \
         libcap-dev libmount-dev libkmod-dev \
-        pkg-config wget python3-jinja2
-
-    # gnu-efi is installed here to enable -Dgnu-efi behind which fuzz-bcd
-    # is hidden. It isn't linked against efi. It doesn't
-    # even include "efi.h" because "bcd.c" can work in "unit test" mode
-    # where it isn't necessary.
-    apt-get install -y gnu-efi zstd
+        pkg-config wget python3-jinja2 zstd

     pip3 install -r .github/workflows/requirements.txt --require-hashes
```

Reply:
I believe this conversation is resolved. I'm not sure I didn't press the button